vars/testsPublishResults.groovy

```diff
@@ -53,6 +55,11 @@ void call(Map parameters = [:]) {
         publishCoberturaReport(configuration.get('cobertura'))
         // PERFORMANCE
         publishJMeterReport(configuration.get('jmeter'))
+        // handle the results
+        if (currentBuild.result == 'UNSTABLE' && configuration.failOnError) {
```

Comment:
who sets the build status to "UNSTABLE" ? Is there a more direct way to judge if there was a failure? In the piper-lib we have the habit of setting the build to unstable in case we deprecate an API in order to create awareness.

What happens if we set the build to unstable. In this case the build will fail here also.

Reply:
Yes, that's true. I think we can access the Test Results on the Jenkins build. But I need to check if we also see the failed ones there.

Reply:
junit plugin sets the build status to `UNSTABLE`.
Since failOnError is by default set to `false` this is OK from my side.

I think we should not invest again in parsing test results since this is the job of the junit plugin  ...

Reply:
Handled this with accessing the test results itself, please check again.